src/components/pages/ImageClassification/ImageAnnotationModal/ImageAnnotationModalMap.js

```diff
@@ -16,44 +15,77 @@ const SELECTED_POLYGON_LINE_WIDTH = 10
 
 const IMAGE_CLASSIFICATION_COLOR_EXP = [
   'case',
+
+  ['get', 'isUnclassified'],
+  COLORS.unclassified,
+
   ['get', 'isConfirmed'],
   COLORS.confirmed,
+
   COLORS.unconfirmed,
 ]
 
-const ImageAnnotationModalMap = ({
-  dataToReview,
-  setDataToReview,
-  highlightedPoints,
-  selectedPoints,
-}) => {
+const getImageScale = (dataToReview) => {
+  const longerSide = Math.max(dataToReview.original_image_width, dataToReview.original_image_height)
+  return longerSide > MAX_DIMENSION ? MAX_DIMENSION / longerSide : 1
+}
+
+const ImageAnnotationModalMap = ({ dataToReview, highlightedPoints, selectedPoints }) => {
   const mapContainer = useRef(null)
   const map = useRef(null)
-  const [dimensions, setDimensions] = useState()
-
-  const _getImageDimensions = useEffect(() => {
-    const imageForMap = new Image()
-    imageForMap.src = dataToReview.image
-    imageForMap.onload = () => {
-      const longerSide = Math.max(imageForMap.naturalWidth, imageForMap.naturalHeight)
-      const imageScale = longerSide > MAX_DIMENSION ? MAX_DIMENSION / longerSide : 1
-
-      setDimensions({
-        height: imageForMap.naturalHeight * imageScale,
-        width: imageForMap.naturalWidth * imageScale,
-        scale: imageScale,
-      })
-    }
-
-    imageForMap.onerror = () => {
-      setDataToReview()
-      toast.error('There was a problem displaying this image, please contact if issue persists.')
-    }
-    // eslint-disable-next-line
-  }, [])
+  const [hasMapLoaded, setHasMapLoaded] = useState(false)
+  const imageScale = getImageScale(dataToReview)
+
+  const getPointsGeojson = useCallback(
+    () => ({
+      type: 'FeatureCollection',
```

Comment:
would suggest adding a check: `if (map.currrent)` at the top of this function or `if (!map.current) return`